### ISSUE TITLE
Rule proposal: no-restricted-paths

### ISSUE THREAD

**Issue by lo1tuma** [benmosher/eslint-plugin-import#155]
Given the following folder structure of a project:

```
my-project
├── client
│   └── foo.js
└── server
    └── bar.js
```

I want to disallow that any file in the `client` folder can import any module from the `server` folder.

The following should be invalid in `client/foo.js`:

```js
import bar from '../server/bar.js';
```

**Comment by benmosher** [comment]
Neat!

How would you propose defining the restricted zone?

I can imagine either regex or a path relative to the eslint working directory, but each is fragile for different reasons.

**Comment by lo1tuma** [comment]
For me relative paths to the cwd of ESLint would work. Do have any specific scenario in mind when relying on the cwd would become a problem?

**Comment by benmosher** [comment]
Not specifically, no. And IME ESLint is traditionally run by npm in the root of a package, so that should be fine. Just thinking it through. 😅

**Comment by sindresorhus** [comment]
An ESLint consumer (plugin/editor/etc) can [define a custom `cwd`](http://eslint.org/docs/developer-guide/nodejs-api#cliengine), so you should not depend on `process.cwd()`. If the set `cwd` isn't exposed in the plugin context, I would open an issue on ESLint to make it so.
